Source/VCGeneration/SplitAndVerifyWorker.cs

```diff
@@ -151,6 +153,44 @@ private async Task ProcessResult(Split split, CancellationToken cancellationToke
       callback.OnProgress?.Invoke("VCprove", splitNumber < 0 ? 0 : splitNumber, total, provenCost / (remainingCost + provenCost));
 
       if (!proverFailed) {
+        if (callback is VerificationResultCollector collector) {
```

Comment:
Relying on `callback` being a specific subtype seems unreliable.

Reply:
Agreed. I think we should put the counter-examples in the VCResult so that I don't need to rely on the collector itself.